include/bio/format/sam_input_handler.hpp

```diff
@@ -0,0 +1,355 @@
+        get<field::tlen>(raw_record)  = (*file_it).fields[8];
+        get<field::seq>(raw_record)   = (*file_it).fields[9];
+        get<field::qual>(raw_record)  = (*file_it).fields[10];
+
+        // fields[10].end() that is guaranteed to be char*
+        char const * end_qual        = (*file_it).fields[10].data() + (*file_it).fields[10].size() + 1 /*\t or \n*/;
+        // line.end() that is guaranteed to be char*
+        char const * end_line        = (*file_it).line.data() + (*file_it).line.size();
+        // SAM tags go from end of qual til end of line (possibly empty)
+        get<field::tags>(raw_record) = std::string_view{end_qual, static_cast<size_t>(end_line - end_qual)};
+    }
+    //!\}
+
+    /*!\name Parsed record handling
+     * \brief This is mostly done via the defaults in the base class.
+     * \{
+     */
+    //!\brief Overload for parsing QNAME.
+    template <typename parsed_field_t>
+    void parse_field(vtag_t<field::qname> const & /**/, parsed_field_t & parsed_field)
+    {
+        std::string_view raw_field = get<field::qname>(raw_record);
+        if (raw_field != "*")
+            parse_field_aux(raw_field, parsed_field); // default parsing
+    }
+
+    //!\brief Overload for parsing FLAG.
+    void parse_field(vtag_t<field::flag> const & tag, map_io::sam_flag & parsed_field)
+    {
+        uint16_t flag_integral{};
+        parse_field(tag, flag_integral); // arithmetic parsing
+        parsed_field = map_io::sam_flag{flag_integral};
+    }
+
+    //!\brief Parse the RNAME field. The reference names are stored in the header.
+    template <typename parsed_field_t>
+    void parse_field(vtag_t<field::rname> const & /**/, parsed_field_t & parsed_field)
+    {
+        std::string_view raw_field = get<field::rname>(raw_record);
+
+        if (raw_field != "*")
+        {
+            size_t rname_pos;
+
+            if (auto it = header.rname_to_pos().find(raw_field); it == header.rname_to_pos().end())
+            { // rname name was not in header, insert!
+                header.push_back_rname(raw_field, raw_field.size(), "");
+
+                rname_pos = header.rnames().size() - 1;
+
+                warning("The reference sequence name \"", raw_field, "\" is not present in the header.");
+            }
+            else
+            {
+                rname_pos = it->second;
+            }
+
+            if constexpr (std::integral<std::remove_cvref_t<parsed_field_t>>)
+                parsed_field = rname_pos;
+            else
+                parsed_field = header.rnames()[rname_pos];
+        }
+    }
+
+    /* POS, MAPQ are handled correctly by default */
+
+    //!\brief Overload for parsing CIGAR.
+    void parse_field(vtag_t<field::cigar> const & /**/, std::vector<seqan3::cigar> & cigar_vector)
+    {
+        std::string_view raw_field = get<field::cigar>(raw_record);
+
+        if (raw_field != "*")
+        {
+            uint32_t     cigar_count{};
```

Comment:
Yes, but then I need to copy over the string.
(currently API is `cigar & assign_string(small_string<11> const s) noexcept`).
This code saves the extra copy.

Reply:
Ok, so I would recommend:
1. Make a TODO at SeqAn3 to change this (easy solution is to just replace it with `std::string_view`. A more elaborate solution could also handle non-contiguous character ranges, but I don't think that's necessarry for now.
2. I would still recommend just calling the interface here for now. It's just 11 bytes copied on the stack, so it shouldn't be to bad. And it will switch automatically once the interface is fixed in SeqAn.